launcher/agent/agent.go

```diff
@@ -114,27 +173,49 @@ func (a *agent) Attest(ctx context.Context, opts AttestAgentOpts) ([]byte, error
 		return nil, fmt.Errorf("failed to get principal tokens: %w", err)
 	}
 
-	var buf bytes.Buffer
-	if err := a.cosCel.EncodeCEL(&buf); err != nil {
-		return nil, err
-	}
-
-	attestation, err := a.attest(challenge.Nonce, buf.Bytes())
+	// attResult can be tdx or tpm or other attest root
+	attResult, err := a.avRot.Attest(challenge.Nonce)
 	if err != nil {
 		return nil, fmt.Errorf("failed to attest: %v", err)
 	}
 
+	var cosCel bytes.Buffer
+	if err := a.avRot.GetCEL().EncodeCEL(&cosCel); err != nil {
+		return nil, err
+	}
+
 	req := verifier.VerifyAttestationRequest{
 		Challenge:      challenge,
 		GcpCredentials: principalTokens,
-		Attestation:    attestation,
 		TokenOptions:   opts.TokenOptions,
 	}
 
+	switch v := attResult.(type) {
+	case *pb.Attestation:
+		a.logger.Info("attestation through TPM quote")
```

Comment:
I think this log line is extraneous. We already have a log line when we select the avRot.

Reply:
right, but I'd prefer to have some extra logging here, might be helpful for debugging later once there are more attestation type.